Mono.Addins/Mono.Addins/AddinLocalizerAttribute.cs

```diff
@@ -61,12 +62,17 @@ public AddinLocalizerAttribute (Type type)
 		/// </summary>
 		public Type Type {
 			get { return type; }
-			set { type = value; typeName = type.FullName; }
+			set { type = value; typeName = type.FullName; assemblyName = type.Assembly.FullName; }
 		}
 
 		internal string TypeName {
 			get { return typeName; }
 			set { typeName = value; type = null; }
 		}
+
+		internal string AssemblyName {
+			get { return assemblyName; }
+			set { assemblyName = value; type = null; }
```

Comment:
The assembly name is not going to be set when using the Cecil reflector. Properties of type "Type" in attributes have a special handling to avoid having to load the actual type in process. This special handling is here: 

https://github.com/mono/mono-addins/blob/master/Mono.Addins.CecilReflector/Mono.Addins.CecilReflector/Reflector.cs#L150. 

When the reflector instantiates the attribute, instead of loading the type and assigning it to the type attribute, it gets the full name of the attribute and assigns it to a "*Name" property. An easy fix is to also try to assign the assembly name to a "*AssemblyName" property. Another option would be to assign to the type name property the fully qualified name of the type, instead of just the full name, and then change the GetType method in RuntimeAddin to optimize lookup for fully qualified type names.